hw4/async/hw4_async.py

```diff
@@ -0,0 +1,80 @@
+'''
+Написать программу, которая скачивает изображения с заданных URL-адресов и сохраняет их на диск. Каждое изображение
+должно сохраняться в отдельном файле, название которого соответствует названию изображения в URL-адресе.
+Например, URL-адрес: https://example/images/image1.jpg -> файл на диске: image1.jpg
+— Программа должна использовать многопоточный, многопроцессорный и асинхронный подходы.
+— Программа должна иметь возможность задавать список URL-адресов через аргументы командной строки.
+— Программа должна выводить в консоль информацию об общем времени выполнения программы.
+Синхронный подход.
+'''
+import argparse
+import asyncio
+import aiohttp
+import aiofiles
+import os
+import time
+
+
+def timer(func):  # декоратор бенчмарк
+	async def wrapper(*args):
+		start_time = time.time()
+		await func(*args)
+		end_time = time.time() - start_time
+		print(f'время выполнения {end_time:.2f} с')
+	return wrapper
+
+
+async def parse(url: str, path):
+	async with aiohttp.ClientSession() as session:
+		async with session.get(url) as response:
+			if response.status == 200:  # если статус ОК
+				name = url.replace('https://', '').split('/')[-1:-2:-1][0]
+				filename = f'{path}//{name}'
```

Comment:
Если поменять имя параметра `path` на `folder` по известным причинам, то чтобы код работал на любой платформе, можно и лучше сделать так,
```suggestion
				filename = os.path.join(folder, name)
```